# Pandas

Nous allons voir quelques astuces pour nettoyer des données.

In [1]:
import pandas as pd

- charger le dataset `retail.csv` :

In [54]:
# Votre code
df = pd.read_csv("retail.csv")
df.head()

BranchName  Week  DayWeek  Day  Month  Hour Transaction_Type  Units   Amount
0    MyStore     1        3    2      1     9             Card      3   $54.00
1    MyStore     1        3    2      1    10             Cash      7  -$17.80
2    MyStore     1        3    2      1    10             Card      7   $41.99
3    MyStore     1        3    2      1    11             Card     20  $412.50
4    MyStore     1        3    2      1    12             Cash      1  -$18.00

In [50]:
df['Amount'][966]

'-$1,041.80'

- utiliser `describe` pour afficher les informations :

In [5]:
# Votre code
df.describe()

Week      DayWeek          Day        Month         Hour  \
count  4100.000000  4100.000000  4100.000000  4100.000000  4100.000000   
mean     34.017805     4.183902    15.812195     8.231463    12.949024   
std      14.714289     1.967864     8.810817     3.396586     2.631853   
min       1.000000     1.000000     1.000000     1.000000     8.000000   
25%      29.000000     3.000000     8.000000     7.000000    11.000000   
50%      37.000000     4.000000    16.000000     9.000000    13.000000   
75%      45.000000     6.000000    23.000000    11.000000    15.000000   
max      53.000000     7.000000    31.000000    12.000000    19.000000   

             Units  
count  4100.000000  
mean     12.779512  
std      17.854968  
min       1.000000  
25%       3.000000  
50%       8.000000  
75%      16.000000  
max     274.000000

- trouver un moyen de checker s'il y a des valeurs nulles sur l'ensemble du dataset (il existe plusieurs façons de faire) :

In [60]:
# Votre code
df.isnull().any()

BranchName          False
Week                False
DayWeek             False
Day                 False
Month               False
Hour                False
Transaction_Type    False
Units               False
Amount              False
dtype: bool

- afficher les types des colonnes :

In [8]:
# Votre code
df.dtypes

BranchName          object
Week                 int64
DayWeek              int64
Day                  int64
Month                int64
Hour                 int64
Transaction_Type    object
Units                int64
Amount              object
dtype: object

- observer la colonne `Amount`. Que constatez-vous ?

In [12]:
# Votre code
df['Amount'].describe()

count       4100
unique      2353
top       -$8.00
freq          28
Name: Amount, dtype: object

Nous allons devoir remplacer ces valeurs par des valeurs numériques afin de pouvoir ensuite nous en servir.

- retirer les `$` et les `,` :

Par exemple : `-$17.80` devient `-17.80`

In [55]:
# Votre code
df['Amount'] = df['Amount'].apply(lambda x: (str(x).replace(",","").replace("$", "")))
df['Amount']

0         54.00
1        -17.80
2         41.99
3        412.50
4        -18.00
5         -5.00
6        118.50
7        -94.92
8        423.50
9        -95.39
10       -16.90
11       452.10
12      -110.40
13      -309.90
14       593.50
15      -377.12
16        77.50
17       -19.50
18       -31.00
19        58.00
20         5.00
21       -51.00
22       -21.30
23       299.00
24       -36.40
25        -3.40
26        17.50
27      -112.80
28       238.00
29       852.50
         ...   
4070      -9.00
4071      76.25
4072     347.50
4073     239.00
4074     -59.81
4075     -36.00
4076     371.00
4077    -102.00
4078     -38.25
4079     388.25
4080     -82.10
4081     -51.00
4082     289.60
4083     -46.00
4084     112.00
4085      -8.00
4086     352.50
4087     -96.60
4088      28.95
4089       4.00
4090     327.00
4091     -33.00
4092     105.50
4093     303.50
4094     -23.21
4095     129.00
4096     -68.36
4097      71.00
4098    -359.18
4099     130.00
Name: Amount, Length: 41

- transformer le type de cette colonne en valeur numérique :

In [57]:
# Votre code
df['Amount'] = pd.to_numeric(df['Amount'])

- checker en réaffichant les types des colonnes :

In [58]:
# Votre code
df.dtypes

BranchName           object
Week                  int64
DayWeek               int64
Day                   int64
Month                 int64
Hour                  int64
Transaction_Type     object
Units                 int64
Amount              float64
dtype: object

- quel est le nombre de vente par mois ?

In [59]:
# Votre code
df.groupby('Month')['Units'].agg(['sum'])

sum
Month      
1      7461
7      7400
8      8024
9      6976
10     6688
11     7730
12     8117